### PR TITLE
[FLINK-21466][table] Add SQL Client default mode value embedded

### PR DESCRIPTION
**Pull request by zck573693104**
1.Add SQL Client default mode value embedded.
2.Modify sqlClient md,support default mode value `embedded`.
3.Test sql client default mode `embedded`.only left a test which uses embedded to verify it still work.

### CODE REVIEW

File: flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/SqlClient.java [hunk 1]
```diff
@@ -125,15 +125,21 @@ private void openCli(String sessionId, Executor executor) {
     // --------------------------------------------------------------------------------------------
 
     public static void main(String[] args) {
-        if (args.length < 1) {
-            CliOptionsParser.printHelpClient();
-            return;
+        final String model;
```
Comment (fsk119) [inline flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/SqlClient.java:128]: rename `model` to `mode`

**Review by fsk119 — CHANGES_REQUESTED** [review]
Thanks for your contribution and sorry for the late response.  I left one minor comments.

**Comment by wuchong** [comment]
I have appended a commit to remove useless `embedded` and also added a test for `SqlClient`. 

Would be great if @fsk119 can have a look again.

**Review by fsk119 — APPROVED** [review]
Thanks Jark's help. The added test is amazing.  

LGTM.

**Comment by wuchong** [comment]
Merging...

**Pull request closed without merging**